core/celo_evm.go

```diff
@@ -9,18 +9,14 @@ import (
 	"github.com/ethereum/go-ethereum/params"
 )
 
-func setCeloFieldsInBlockContext(blockContext *vm.BlockContext, header *types.Header, config *params.ChainConfig, statedb vm.StateDB) {
-	if !config.IsCel2(header.Time) {
-		return
-	}
```

Comment:
Shouldn't we keep this with a `return nil` or add a statement like it to `contract.GetFeeCurrencyContext`?

Reply:
Ok, shortcut.
I will return an empty instance. The block context doesn't store a pointer, and it was working before assuming that it would be created in the context anyways. So to avoid bugs, an empty structure would be the same as before